File: 2018/02_py/02.py
from collections import Counter
import sys

boxes = []

with open('02_input.txt') as file:
    boxes = list(map(lambda line: line.strip(), file.readlines()))

def part1():
    has2 = 0
    has3 = 0

    for count in [Counter(box).values() for box in boxes]:
        has2 += 2 in count
        has3 += 3 in count

    checksum = has2 * has3

    print(checksum)

def part2():
    pass

part1()
part2()

File: 2018/02_js/02.js
import fs from 'fs';

function Counter(s) {
    var count = {};
    Array.from(s).forEach(val => count[val] = (count[val] || 0) + 1);
    return count;
}

fs.readFile('02_input.txt', 'utf8', function (err, contents) {
    const boxes = contents.split('\n').map(box => box.trim());

    const counts = boxes.map(box => Counter(box)).map(count => {
        return Object.values(count);
    })

    let has2 = 0
    let has3 = 0

    counts.forEach(count => {
        if (count.includes(2)) {
            has2++;
        }

        if (count.includes(3)) {
            has3++;
        }
    });

    const checksum = has2 * has3;

    console.log(checksum)
});